Fix #40067: Add dedicated UMT5 support to GGUF loader (config, tokenizer, test)

akshay-babbar opened the pull request:
### PR Description

### Problem

**Fixes #40067**

UMT5EncoderModel.from_pretrained() fails when loading GGUF files, defaulting to T5 configuration and producing the warning:
```
You are using a model of type t5 to instantiate a model of type umt5. This is not supported for all configurations of models and can yield errors.
```

### Solution

Adds dedicated UMT5 architecture support to the GGUF loader:
- **Detection**: Identifies UMT5 models by name pattern in GGUF files
- **Configuration**: Maps GGUF parameters to UMT5Config instead of falling back to T5
- **Integration**: Registers UMT5 with tokenizer conversion and architecture mapping

### Changes

- `src/transformers/modeling_gguf_pytorch_utils.py`: Added UMT5 model detection and config/architecture mapping
- `src/transformers/integrations/ggml.py`: Added UMT5 config mapping and tokenizer support
- `tests/models/t5/test_modeling_t5.py`: Added integration test for UMT5 GGUF loading

### Testing
✅ New integration test validates UMT5EncoderModel loads correctly from GGUF
✅ Verifies proper config type (UMT5Config, not T5Config)
✅ Confirms weights map without tensor mismatches
✅ Ensures forward pass returns valid last_hidden_state

### Impact
Enables users to load UMT5 encoder models from GGUF format without configuration warnings or errors.
This is not only a warning fix — it ensures GGUF UMT5 models load into the correct architecture + tokenizer, preventing subtle runtime errors.

**Scope**: This patch adds GGUF support for UMT5 encoder models only. Decoder/Seq2Seq GGUF support is out of scope and can be addressed in a separate follow-up.

**Before**: ❌
UMT5EncoderModel.from_pretrained("city96/umt5-xxl-encoder-gguf", gguf_file="...")  # fails
**After**: ✅
UMT5EncoderModel.from_pretrained("city96/umt5-xxl-encoder-gguf", gguf_file="...")  # works correctly


Thanks!

akshay-babbar committed:
Fix #40067 : add UMT5 support in GGUF loader (config, tokenizer, test)

akshay-babbar committed:
chore: fix code formatting and linting issues

akshay-babbar commented:
Hi @MekkCyber , just wanted to check in if you’ve had a chance to review this PR. Thanks!

MekkCyber reviewed (commented):
Thanks a lot @akshay-babbar for this integration ! It looks great just left some nits

akshay-babbar committed:
refactor: move UMT5 GGUF test to quantization directory and clean up comments

akshay-babbar commented on src/transformers/modeling_gguf_pytorch_utils.py:
```diff
@@ -386,9 +388,15 @@ def load_gguf_checkpoint(gguf_checkpoint_path, return_tensors=False, model_to_lo
-            parsed_parameters["config"]["architectures"] = ["T5EncoderModel"]
-        updated_architecture = "t5"
+        # fix-umt5-gguf-config-support
+        if model_name and "umt5" in model_name[0].lower():
```
removed!

akshay-babbar commented on tests/models/t5/test_modeling_t5.py:
```diff
@@ -1855,3 +1856,38 @@ def test_defaulting_to_symmetry(self):
+        gguf_path = "umt5-xxl-encoder-Q8_0.gguf"
+        model = UMT5EncoderModel.from_pretrained(
+            model_id,
+            gguf_file=gguf_path,
```
done!

akshay-babbar commented:
Hi @MekkCyber , just wanted to check in if you’ve had a chance to review my changes? Thanks!

akshay-babbar committed:
chore: trigger CI pipeline

akshay-babbar committed:
Merge branch 'main' into fix-umt5-gguf-config-support

akshay-babbar commented:
Hi @MekkCyber, I understand you might be busy, no worries 🙂
@Rocketknight1 / @ArthurZucker , would you (or someone you recommend) be able to help merge this PR? All tests are passing, and I’ve incorporated the suggestions above🙏 

Thanks!

MekkCyber commented on tests/quantization/ggml/test_ggml.py:
```diff
@@ -1072,3 +1080,38 @@ def test_qwen3moe_q4_k_m(self):
+
+@slow
+@require_torch
+class T5GGUFIntegrationTests(unittest.TestCase):
+    def test_umt5_encoder_can_load_gguf_from_hub_path(self):
+        """
+        Verifies that a UMT5 encoder loads directly from a GGUF file using
+        UMT5EncoderModel.from_pretrained(...), and the config is correctly UMT5.
+        """
```
we should not have a specific class for each model, all model tests should be under `GgufModelTests`

MekkCyber reviewed (commented):
Sorry for the delay @akshay-babbar, I left one small nit and good to go

akshay-babbar committed:
refactor(tests): Move UMT5 Encoder GGUF test to GgufModelTests. This consolidates the new test into the main class for consistency.

MekkCyber reviewed (commented):
Thanks a lot @akshay-babbar, one minor comment to take care of

akshay-babbar committed:
Add regression check to UMT5 encoder GGUF test

Verify encoder output against reference tensor values with appropriate tolerances for stability.

akshay-babbar committed:
Merge branch 'main' into fix-umt5-gguf-config-support

HuggingFaceDocBuilderDev commented:
The docs for this PR live [here](https://moon-ci-docs.huggingface.co/docs/transformers/pr_40218). All of your documentation changes will be reflected on that endpoint. The docs are available until 30 days after the last update.

MekkCyber commented on tests/quantization/ggml/test_ggml.py:
```diff
@@ -1072,3 +1081,41 @@ def test_qwen3moe_q4_k_m(self):
+        # architectures should reflect the encoder variant
+        self.assertIn("UMT5EncoderModel", getattr(model.config, "architectures", []))
+
+        # --- Smoke: tiny forward pass to ensure weights/tensor mapping are valid ---
+        input_ids = torch.tensor([[1, 2, 3, 4]], dtype=torch.long).to(torch_device)
```
```suggestion
        input_ids = torch.tensor([[1, 2, 3, 4]], dtype=torch.long).to(torch_device)
```

MekkCyber reviewed (approved):
Thanks @akshay-babbar ! LGTM ! let's just remove the comments below and we are good to go

akshay-babbar committed:
Update tests/quantization/ggml/test_ggml.py

Co-authored-by: Mohamed Mekkouri <93391238+MekkCyber@users.noreply.github.com>

akshay-babbar committed:
Update tests/quantization/ggml/test_ggml.py

remove comments

Co-authored-by: Mohamed Mekkouri <93391238+MekkCyber@users.noreply.github.com>

MekkCyber committed:
Merge branch 'main' into fix-umt5-gguf-config-support

MekkCyber committed:
Merge branch 'main' into fix-umt5-gguf-config-support

akshay-babbar committed:
Merge branch 'main' into fix-umt5-gguf-config-support

MekkCyber committed:
Merge branch 'main' into fix-umt5-gguf-config-support

MekkCyber committed:
Merge branch 'main' into fix-umt5-gguf-config-support

MekkCyber committed:
Merge branch 'main' into fix-umt5-gguf-config-support

akshay-babbar committed:
Merge branch 'main' into fix-umt5-gguf-config-support

MekkCyber committed:
Merge branch 'main' into fix-umt5-gguf-config-support

MekkCyber committed:
Merge branch 'main' into fix-umt5-gguf-config-support

MekkCyber commented:
@akshay-babbar it's not related, just a ci problem it seems
